docs/source/howto/queryable-encryption.rst

```diff
@@ -0,0 +1,155 @@
+        ),
+        "encrypted": parse_uri(
+            DATABASE_URL,
+            options: {
+                "auto_encryption_opts": AutoEncryptionOpts(
+                    key_vault_namespace="my_encrypted_database.keyvault",
+                    kms_providers={"local": {"key": os.urandom(96)}},
+                ),
+            },
+            db_name="my_encrypted_database",
+            "KMS_CREDENTIALS": {},
+        ),
+    }
+
+    class EncryptedRouter:
+        """
+        A database router for an encrypted database to be used with a
+        "patientdata" app that contains models with encrypted fields.
+        """
+
+        def allow_migrate(self, db, app_label, model_name=None, **hints):
+            # The patientdata app's models are only created in the encrypted database.
+            if app_label == "patientdata":
+                return db == "encrypted"
+            # Don't create other app's models in the encrypted database.
+            if db == "encrypted":
+                return False
+            return None
+
+        def kms_provider(self, model, **hints):
+            return "local"
+
+    DATABASE_ROUTERS = [EncryptedRouter()]
+
+.. admonition:: KMS providers and credentials
+
+    The above example uses a local KMS provider with a randomly generated
+    key. In a production environment, you should use a secure KMS provider
+    such as AWS KMS, Azure Key Vault, or GCP KMS.
+
+    Please refer to :ref:`manual:qe-fundamentals-kms-providers`
+    for more information on configuring KMS providers and credentials as well as
+    :doc:`manual:core/queryable-encryption/fundamentals/keys-key-vaults`
+    for information on creating and managing data encryption keys.
+
+    You can also refer to the `Python Queryable Encryption Tutorial
+    <https://github.com/mongodb/docs/tree/adad2b1ae41ec81a6e5682842850030813adc1e5/source/includes/qe-tutorials/python>`_.
+
+Encrypted fields map
+~~~~~~~~~~~~~~~~~~~~
+
+In addition to the :ref:`settings described in the how-to guide
+<queryable-encryption-settings>` you will need to provide a
+``encrypted_fields_map`` to the ``AutoEncryptionOpts``.
+
+You can use the :djadmin:`showencryptedfieldsmap` management command to generate
+the schema map for your encrypted fields and then use the results in your
+settings::
+
+    python manage.py showencryptedfieldsmap --database=encrypted
+
+.. admonition:: Didn't work?
+
+    If you get the error ``Unknown command: 'showencryptedfieldsmap'``, ensure
+    ``"django_mongodb_backend"`` is in your :setting:`INSTALLED_APPS` setting.
+
+Crypt shared library
+~~~~~~~~~~~~~~~~~~~~
+
+Additionally, you will need to ensure that the :ref:`crypt shared library is
+available <manual:qe-reference-shared-library>` to your Python environment.
+The crypt shared library is recommended over mongocryptd for Queryable
+Encryption because it doesn't require an additional daemon process to run to
+facilitate Queryable Encryption operations.
```

Comment:
Isn't this more like a prerequisite? In the previous step, you ran `showencryptedfieldsmap`. Doesn't that require encryption? (I installed enterprise which includes mongocryptd, so I haven't needed the shared library, which perhaps will be a point of confusion)

Reply:
Maybe, see: https://github.com/mongodb/django-mongodb-backend/pull/329#issuecomment-3160503159